### PR TITLE
[feature] piv: manage trip addition

### PR DESCRIPTION
**Pull request by patochectp**
ref: ND-1021

### CODE REVIEW

**Review by woshilapin — APPROVED** [review]

File: tests/integration/piv_test.py [hunk 1]
```diff
@@ -1661,3 +1658,226 @@ def test_piv_re_routed_in_the_end(mock_rabbitmq):
...
+        assert added_st.departure == datetime(2020, 10, 22, 21, 25)
+
+        added_st = db_trip.stop_time_updates[5]
+        assert added_st.arrival_status == ModificationType.add.name
+        assert added_st.arrival == datetime(2020, 10, 22, 21, 30)
+        assert added_st.departure_status == ModificationType.none.name
+        assert added_st.departure == datetime(2020, 10, 22, 21, 30)
+
+
+def test_piv_trip_creation_by_ope_or_ptp(mock_rabbitmq):
```
Comment (pbougue) [inline tests/integration/piv_test.py:1780]: :+1: Maybe we should check the "PTP" version, to be sure it never disappear from connector.

File: tests/integration/piv_test.py [hunk 2]
```diff
@@ -1661,3 +1658,226 @@ def test_piv_re_routed_in_the_end(mock_rabbitmq):
...
+        assert st.arrival_status == ModificationType.add.name
+        assert st.arrival == datetime(2020, 10, 22, 21, 25)
+        assert st.departure_status == ModificationType.none.name
+        assert st.departure == datetime(2020, 10, 22, 21, 25)
+
+
+def test_piv_trip_creation_wrong_company_and_physical_mode(mock_rabbitmq):
+    piv_feed = _get_stomp_20201022_23188_add_trip_fixture()
+    piv_feed["objects"][0]["object"]["operateur"]["codeOperateur"] = "1180"
+    piv_feed["objects"][0]["object"]["modeTransport"]["typeMode"] = "FEROUTIER"
```
Comment (pbougue) [inline tests/integration/piv_test.py:1858]: :truck::train: :grin:

File: kirin/piv/model_maker.py [hunk 3]
```diff
@@ -407,20 +412,24 @@ def _make_trip_update(self, json_train, vj):
...
+                    if piv_event_status in ["SUPPRESSION_PARTIELLE"]:
                         setattr(st_update, STATUS_MAP[event_toggle], ModificationType.delete.name)
                     elif piv_event_status in ["SUPPRESSION_DETOURNEMENT"]:
                         setattr(st_update, STATUS_MAP[event_toggle], ModificationType.deleted_for_detour.name)
                     elif piv_event_status in ["CREATION"]:
                         setattr(st_update, STATUS_MAP[event_toggle], ModificationType.add.name)
                     elif piv_event_status in ["CREATION_DETOURNEMENT"]:
                         setattr(st_update, STATUS_MAP[event_toggle], ModificationType.added_for_detour.name)
+                    elif trip_status_type == TripEffect.ADDITIONAL_SERVICE:
+                        setattr(st_update, STATUS_MAP[event_toggle], ModificationType.add.name)
```
[the remark this answers is not in the document]
Comment (patochectp) [reply kirin/piv/model_maker.py:430]: done in 6665a231ecc10a037782c75a32e69bb741888fd4

File: tests/integration/piv_test.py [hunk 4]
```diff
@@ -1661,3 +1658,226 @@ def test_piv_re_routed_in_the_end(mock_rabbitmq):
...
+        db_trip = TripUpdate.query.first()
+
+        assert db_trip.status == ModificationType.add.name
+        assert db_trip.effect == TripEffect.ADDITIONAL_SERVICE.name
+        assert len(db_trip.stop_time_updates) == 6
+
+        added_st = db_trip.stop_time_updates[0]
+        assert added_st.arrival_status == ModificationType.none.name
+        assert added_st.arrival == datetime(2020, 10, 22, 20, 34)
+        assert added_st.departure_status == ModificationType.add.name
```
[the remark this answers is not in the document]
Comment (patochectp) [reply tests/integration/piv_test.py:1676]: done in 6665a231ecc10a037782c75a32e69bb741888fd4

File: tests/integration/piv_test.py (hunk 1 above)
Comment (patochectp) [reply tests/integration/piv_test.py:1799]: done in 6665a231ecc10a037782c75a32e69bb741888fd4

File: kirin/piv/model_maker.py [hunk 5]
```diff
@@ -335,7 +339,7 @@ def _get_navitia_vj(self, piv_key, ads):
             navitia_vj = navitia_vjs[0]
             try:
                 base_vs_rt_error_margin = timedelta(hours=1)
-                vj_base_start = _get_first_stop_base_datetime(ads, "depart", skip_fully_added_stops=True)
+                vj_base_start = _get_first_stop_base_datetime(ads, "depart", skip_fully_added_stops=False)
```
[the remark this answers is not in the document]
Comment (patochectp) [reply kirin/piv/model_maker.py]: update in 6665a231ecc10a037782c75a32e69bb741888fd4

File: kirin/piv/model_maker.py [hunk 6]
```diff
@@ -304,14 +308,15 @@ def build_trip_updates(self, rt_update):
         # replace by cleaned and sorted version of stoptimes list.
         json_train["listeArretsDesserte"]["arret"] = ads
 
-        vj = self._get_navitia_vj(piv_key, ads)
+        trip_addition = higher_disruption.get("type") == "CREATION"
```
Comment (pbougue) [inline kirin/piv/model_maker.py:311]: Suggesting 2 remanes:
* As bool are mostly named like that,
* As I always have :cold_sweat: thinking disruption is the higher among stops
```suggestion
        is_trip_addition = higher_trip_disruption.get("type") == "CREATION"
```

File: tests/integration/piv_test.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (pbougue) [reply tests/integration/piv_test.py:1676]: My point was also to add `assert`s in this fn

File: tests/integration/piv_test.py [hunk 7]
```diff
@@ -1661,3 +1658,245 @@ def test_piv_re_routed_in_the_end(mock_rabbitmq):
...
+    assert "PIV feed processed" in res.get("message")
+
+    _assert_db_piv_trip_creation()
+
+
+def test_piv_trip_creation_delayed_0min(mock_rabbitmq):
+    piv_feed = _get_stomp_20201022_23188_add_trip_fixture()
+    disruption = DisruptionTuple(type="RETARD_PROJETE", texte="Absence inopinée d'un agent")
+    modification = ModificationTuple(motif=None, statut=None)
+    retard = {"duree": 0, "dureeInterne": 3}
```
Comment (pbougue) [inline tests/integration/piv_test.py:1722]: Can you change it to have a case which is more tricky?
```suggestion
    retard = {"duree": 5, "dureeInterne": 6}
```
This should still give a `0` delay as this is added stop, so only `dateHeureReelle` is considered.

Comment (patochectp) [reply tests/integration/piv_test.py:1722]: It conforms with test_piv_trip_creation_delayed_5min test

**Review by pbougue — APPROVED** [review]

**Review by pbougue — APPROVED** [review]

**Review by pbougue — APPROVED** [review]

**Pull request merged**